libs/common/src/lib/resources/resource-list/resource-list.service.ts

```diff
@@ -0,0 +1,273 @@
+import { inject, Injectable } from '@angular/core';
+import { SDKService, StateService } from '@flaps/core';
+import { BehaviorSubject, catchError, forkJoin, Observable, of, Subject, switchMap, take, tap } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { DEFAULT_PAGE_SIZE, DEFAULT_SORTING, ResourceWithLabels } from './resource-list.model';
+import {
+  IResource,
+  LabelSetKind,
+  LabelSets,
+  ProcessingStatusResponse,
+  Resource,
+  RESOURCE_STATUS,
+  Search,
+  SearchOptions,
+  SortOption,
+} from '@nuclia/core';
+import { TranslateService } from '@ngx-translate/core';
+import { LabelsService } from '../../label/labels.service';
+import { UploadService } from '../../upload/upload.service';
+
+@Injectable({ providedIn: 'root' })
+export class ResourceListService {
+  private sdk = inject(SDKService);
+  private labelService = inject(LabelsService);
+  private translate = inject(TranslateService);
+  private stateService = inject(StateService);
+  private uploadService = inject(UploadService);
+
+  private processingStatus?: ProcessingStatusResponse;
+  private _status: RESOURCE_STATUS = RESOURCE_STATUS.PROCESSED;
+  get status() {
+    return this._status;
+  }
+  set status(status: RESOURCE_STATUS) {
+    this._status = status;
+    if (status === RESOURCE_STATUS.PENDING && !this.sdk.nuclia.options.standalone) {
+      forkJoin([this.stateService.account.pipe(take(1)), this.stateService.kb.pipe(take(1))])
+        .pipe(
+          filter(([account, kb]) => !!account && !!kb),
+          take(1),
+          switchMap(([account]) => this.sdk.nuclia.db.getProcessingStatus(account!.id)),
+        )
+        .subscribe((processingStatus) => (this.processingStatus = processingStatus));
+    }
+  }
+
+  private _filters = new BehaviorSubject<string[]>([]);
+  filters = this._filters.asObservable();
+  isFiltering = this.filters.pipe(map((filters) => filters.length > 0));
+
+  private _ready = new BehaviorSubject<boolean>(false);
+  ready = this._ready.asObservable();
+  private _data = new BehaviorSubject<ResourceWithLabels[]>([]);
+  data = this._data.asObservable();
+  private _emptyKb = new Subject<boolean>();
+  emptyKb = this._emptyKb.asObservable();
+
+  labelSets: Observable<LabelSets> = this.labelService.getLabelsByKind(LabelSetKind.RESOURCES).pipe(
+    filter((labelSets) => !!labelSets),
+    map((labelSets) => labelSets as LabelSets),
+  );
+
+  private _page = 0;
+  private _hasMore = false;
+  private _pageSize = DEFAULT_PAGE_SIZE;
+  private _sort: SortOption = DEFAULT_SORTING;
+  private _query = '';
+  private _titleOnly = true;
+
+  clear() {
+    this._ready.next(false);
+    this._data.next([]);
+    this._page = 0;
+    this._hasMore = false;
+    this._pageSize = DEFAULT_PAGE_SIZE;
+    this._sort = DEFAULT_SORTING;
+    this._query = '';
+    this._titleOnly = true;
+  }
+
+  get sort(): SortOption {
+    return this._sort;
+  }
+
+  filter(filters: string[]) {
```

Comment:
Not a big fan. When a service makes a subscribe in its methods, it is very error prone: when creating a new component we might call it in a subscribe without being aware. The developer has to carefully check how the service method is implemented because the typing will not indicate anything.

2 solutions:
- no subscribe: the method returns an observable and the component is meant to subscribe itself (that's super explicit, typing will indicate that's an observable, but I agree it can be boring and produce more code on the component side)
- decoupling: add in the service a private `_triggerResourceLoad` subject, in constructor, subscribe to the subject to call `loadResource`, and in the `filter` method (and all the other ones subscribing to `loadResource` at the moment), you just do `this._triggerResourceLoad.next()`.
That way the method is entirely safe to be called from wherever (in a subscribe or not).